api/adapters/hydroshare.py

```diff
@@ -45,23 +48,23 @@ def to_dataset_creator(self):
 
 class Award(BaseModel):
     funding_agency_name: str
-    title: Optional[str]
-    number: Optional[str]
-    funding_agency_url: Optional[HttpUrl]
+    title: Optional[str] = None
+    number: Optional[str] = None
+    funding_agency_url: Optional[HttpUrl] = None
 
     def to_dataset_grant(self):
-        grant = schema.Grant.construct()
+        grant = schema.Grant.model_construct()
         if self.title:
             grant.name = self.title
         else:
             grant.name = self.funding_agency_name
         if self.number:
             grant.identifier = self.number
 
-        funder = schema.Organization.construct()
+        funder = schema.Organization.model_construct()
         funder.name = self.funding_agency_name
         if self.funding_agency_url:
-            funder.url = self.funding_agency_url
+            funder.url = str(self.funding_agency_url)
```

Comment:
Should the Organization.funder.url attribute be updated to an HttpUrl instead of casting to a string here?

Reply:
The funder url has been updated to type HttpUrl:
https://github.com/I-GUIDE/catalogapi/blob/38-update-to-pydantic-v2/api/adapters/hydroshare.py#L53